### PR TITLE
optimized Gaussian io functions

### PR DESCRIPTION
**Pull request by Franklalalala**
1. the gau conventions are enhanced with QH9 dataset related and some other conventions. The old filename has a typo, but has been fixed.
2. the two gau.log are changed to Key word IOp(5/33=3, 3/33=1), which has the correct Fock matrix.
3. add many tool functions
4. the gau parser is refactored for better readability

This PR has been tested with cmd line usage with the two updated log files.

### CODE REVIEW

File: dftio/io/gaussian/gaussian_conventions.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+orbital_sign_map = {
+    's': [1],
+    'p': [-1, 1, -1],
+    'd': [1, -1, 1, -1, 1],
+    'f': [-1, 1, -1, 1, -1, 1, -1],
+}
+
+gau_6311_plus_gdp_convention = {'atom_to_simplified_orbitals': {'C': 'sspspspspd', 'H': 'sssp', 'O': 'sspspspspd'},
+                                'atom_to_sorted_orbitals': {'C': 'sssssppppd', 'H': 'sssp', 'O': 'sssssppppd'},
+                                'atom_to_transform_indices': {'C': [0, 1, 5, 9, 13, 3, 4, 2, 7, 8, 6, 11, 12,
```
Comment (floatingCatty) [inline dftio/io/gaussian/gaussian_conventions.py:28]: Is these map suggest we only support these elements?

Comment (Franklalalala) [reply dftio/io/gaussian/gaussian_conventions.py:28]: Yes! It's tailored for specific datasets. See [QH9](https://github.com/divelab/AIRS/blob/502422e538ad274925fda83f6795341643e31368/OpenDFT/QHBench/QH9/datasets.py#L23)

Comment (Franklalalala) [reply dftio/io/gaussian/gaussian_conventions.py:28]: However, by default, we are calculating convention files on the fly. In this case, the [gaussian_conventions.py](https://github.com/deepmodeling/dftio/pull/6/files/2930f91dc034009c4f6d917b2847765e76f20a6f#diff-06c3c7bb29026bad6bb4b958a0ec7ce523e5d283d57adb3b44b36e3a91b2ad5e) will not be used.

Comment (Franklalalala) [reply dftio/io/gaussian/gaussian_conventions.py:28]: Here, by defalut, the convention_file=None. In this case, the element requirement is not strict.
PixPin_2024-11-21_01-23-42

**Comment by Franklalalala** [comment]
changed typo in code

**Pull request merged**